src/common/module_builder.h

```diff
@@ -3,40 +3,73 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#ifndef IREE_PJRT_PLUGIN_PJRT_COMMON_MODULE_BUILDER_H_
-#define IREE_PJRT_PLUGIN_PJRT_COMMON_MODULE_BUILDER_H_
+#ifndef TT_XLA_SRC_COMMON_MODULE_BUILDER_H_
+#define TT_XLA_SRC_COMMON_MODULE_BUILDER_H_
 
+// c++ standard library includes
 #include <memory>
 #include <string>
 
+// llvm mlir includes
 #include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/MLIRContext.h"
 #include "mlir/IR/OwningOpRef.h"
-#include "tt/runtime/runtime.h"
+
+// tt-xla includes
+#include "status.h"
 
 namespace tt::pjrt {
 
 class ModuleBuilder {
 public:
-  ModuleBuilder() = default;
-  ~ModuleBuilder() = default;
+  ModuleBuilder();
+
+  tt_pjrt_status buildModule(const std::string_view &code,
+                             const std::string_view &format);
 
-  size_t get_num_inputs() { return num_inputs_; };
-  size_t get_num_outputs() { return num_outputs_; };
-  unsigned int get_code_size() { return code_size_; };
+  std::shared_ptr<void> getBinary() const { return m_flatbuffer_binary; }
 
-  std::shared_ptr<void> GetBinary() { return binary_ptr_; }
+  size_t getNumInputs() const { return m_num_inputs; };
 
-  void BuildModule(std::string_view code, std::string_view format,
-                   mlir::MLIRContext &context);
+  size_t getNumOutputs() const { return m_num_outputs; };
 
 private:
-  size_t num_inputs_ = 0;
-  size_t num_outputs_ = 0;
-  unsigned int code_size_ = 0;
-  std::shared_ptr<void> binary_ptr_;
-  std::unique_ptr<tt::runtime::Binary> binary_;
+  // Creates VHLO module from the input program code.
+  mlir::OwningOpRef<mlir::ModuleOp>
+  createVHLOModule(const std::string_view &code);
+
+  // Converts VHLO module to StableHLO module.
+  void convertFromVHLOToSHLO(mlir::OwningOpRef<mlir::ModuleOp> &mlir_module);
+
+  // Converts StableHLO module to TTIR module.
+  void convertFromSHLOToTTIR(mlir::OwningOpRef<mlir::ModuleOp> &mlir_module);
+
+  // Converts TTIR module to TTNN module.
+  void convertFromTTIRToTTNN(mlir::OwningOpRef<mlir::ModuleOp> &mlir_module);
+
+  // Creates flatbuffer binary from the built TTNN module.
+  void
+  createFlatbufferBinary(const mlir::OwningOpRef<mlir::ModuleOp> &mlir_module);
```

Comment:
Sorry but I don't think that it should follow the same naming, `convert` functions are converting module in place from one dialect to the other, while `createFlatbufferBinary` creates something new (flatbuffer binary). I could've named it `createFlatbufferBinaryFromTTNNModule` but I feel like it's just needlessly long and there is no ambiguity since we are creating flatbuffer binaries only from TTNN modules. Let me know if you disagree.